sklearn/metrics/_classification.py

```diff
@@ -141,6 +145,27 @@ def _weighted_sum(sample_score, sample_weight, normalize=False):
         return sample_score.sum()
 
 
+def _nan_average(scores: np.ndarray, weights: Optional[np.ndarray]):
+    """
+    Wrapper for np.average, with np.nan values being ignored from the average
+    This is similar to np.nanmean, but allowing to pass weights as in np.average
```

Comment:
submitted an issue to numpy for this: https://github.com/numpy/numpy/issues/21375, but let me know if there's a better solution than this wrapper!

Reply:
What is the performance difference for running `_nan_average` and `np.average` when there are no nans?

Reply:
idk... but I updated this so that if no weights are passed it does just np.nanmean which should be very fast. Otherwise, it creates the mask and does the extra work. tbh, I cannot foresee a big degradation since the operations here are very quick, but 🤷‍♂️